src/components/structures/login/Login.js

```diff
@@ -218,8 +233,12 @@ module.exports = React.createClass({
         }).done();
     },
 
-    onUsernameChanged: function(username) {
+    onUsernameChanged: function(username, endOfInput) {
```

Comment:
I would be tempted to just have a separate `onUsernameBlur` rather than adding a param